pkg/scheduler/objects/application.go

```diff
@@ -1387,6 +1386,17 @@ func (sa *Application) getPlaceholderAllocations() []*Allocation {
 	return allocations
 }
 
+// get a copy of all requests of the application
+func (sa *Application) GetAllRequests() []*AllocationAsk {
+	sa.RLock()
+	defer sa.RUnlock()
+	var requests []*AllocationAsk
+	for _, req := range sa.requests {
+		requests = append(requests, req)
+	}
+	return requests
+}
```

Comment:
Code duplication we already have an unlocked `getAllReqeusts()` This locked version should be a wrapper around the unlocked version.

Reply:
Fixed, also renamed to `getAllRequestsInternal()` to match `addAllocationInternal()`.